src/map/mob.cpp

```diff
@@ -1413,31 +1413,91 @@ static int32 mob_ai_sub_hard_lootsearch(struct block_list *bl,va_list ap)
 }
 
 static int32 mob_warpchase_sub(struct block_list *bl,va_list ap) {
-	struct block_list *target;
-	struct npc_data **target_nd;
-	struct npc_data *nd;
+	block_list *target;
+	block_list **target_warp;
 	int32 *min_distance;
 	int32 cur_distance;
 
 	target= va_arg(ap, struct block_list*);
-	target_nd= va_arg(ap, struct npc_data**);
+	target_warp= va_arg(ap, struct block_list**);
 	min_distance= va_arg(ap, int32*);
 
-	nd = (TBL_NPC*) bl;
+	switch (bl->type) {
+		case BL_NPC:
+		{
+			// NPC Warp
+			npc_data* nd = reinterpret_cast<npc_data*>(bl);
 
-	if(nd->subtype != NPCTYPE_WARP)
-		return 0; //Not a warp
+			// Not a warp
+			if (nd->subtype != NPCTYPE_WARP)
+				return 0;
 
-	if(nd->u.warp.mapindex != map_getmapdata(target->m)->index)
-		return 0; //Does not lead to the same map.
+			// Does not lead to the same map as target
+			if (nd->u.warp.mapindex != map_getmapdata(target->m)->index)
+				return 0;
 
-	cur_distance = distance_blxy(target, nd->u.warp.x, nd->u.warp.y);
-	if (cur_distance < *min_distance)
-	{	//Pick warp that leads closest to target.
-		*target_nd = nd;
-		*min_distance = cur_distance;
-		return 1;
+			// Leads to a different map that is set to not be accessible through warp chase
+			int16 warp_m = map_mapindex2mapid(nd->u.warp.mapindex);
```

Comment:
Check for the battle_config first and then do this lookup inside an if, only if necessary.

Reply:
Done